### PR TITLE
Added some rough guidelines to describe the new PR/commit workflow.

### PR DESCRIPTION
**Pull request by andysim**
This will need some additions eventually, but may be useful for developers that are not familiar with our new workflow.

### CODE REVIEW

**Comment by dgasmith** [comment]
Going to let this test out as another check on wavefunction_pass then ill merge it.

**Comment by andysim** [comment]
Sounds good to me.  I just wanted to get some rough guidelines in there, now that we've made master private.  Feel free to add/remove anything.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+* Add some really awesome code.  It's usually a good idea to make changes on
+  a new branch, but that is not mandatory.
+* If it's a new feature, you must add test cases and documentation.
+* When the code is ready to go, make sure you run the full test suite to make
+  sure nothing was broken.
+* If all tests pass, you can push the code to your fork of Psi4 on GitHub.  If
+  you signed up for a Travis CI account and linked it to your GitHub profile,
+  some rudimentary tests will automatically start.
+* Sign in to GitHub, navigate to your fork of Psi4 and submit a pull request.
+* If the tests all pass, on of the Psi4 admin team will merge the code into the
```
Comment (jturney) [inline CONTRIBUTING.md]: Should this be?: ```If the tests all pass, someone on the Psi4 admin team...```

**Comment by andysim** [comment]
Oops.  My bad.  I'll wait for more substantive changes are made before fixing.  Good catch.

**Comment by loriab** [comment]
Roberto has this https://github.com/PCMSolver/pcmsolver/blob/release_1.1.0/CONTRIBUTING.md that we may want to consider or at least link to.

File: CONTRIBUTING.md (hunk 1 above)
Comment (dgasmith) [inline CONTRIBUTING.md]: You might also want to add the use of @psi4/p4pub-rw tag to alert everyone with psi4/psi4 PR access.

**Comment by loriab** [comment]
Agree with it all, except the bit suggesting ppl oughtn't push to their own fork until changes are nearly ready for upstream merge. May want to link with the wiki: https://github.com/psi4/psi4/wiki/1_Obtaining#fork-from-public-github-repository

**Comment by jgonthier** [comment]
These guidelines seem clear enough to me. Thanks!

**Comment by andysim** [comment]
Thanks for the feedback, all.  I agree that we can recommend people launch a PR early on, and only when it's marked as ready to go will it be pulled in.  To that end, I'll mess with the [new PR templates](https://github.com/blog/2111-issue-and-pull-request-templates) that GitHub just launched; a little checkbox would be a nice, clean solution.

**Comment by andysim** [comment]
I incorporated all of your suggestions, but feel free to edit any of these files further.  The various .md files can now be moved to a .github folder in root, if you're concerned about clutter.  As far as I'm concerned, this is ready to merge.

**Comment by loriab** [comment]
Looks good to me. Thanks, Andy

**Comment by dgasmith** [comment]
This looks awesome! Lets wait until this evening before a merge for additional comments, and I will find someone new to try this out on.

**Comment by dgasmith** [comment]
I think [this](http://codeinthehole.com/writing/pull-requests-and-other-good-practices-for-teams-using-github/) article is the best summary I have found of why we use/recommend/setup github this way. Does someone have a better one?

**Comment by andysim** [comment]
I haven't seen any articles advocating the new workflow, but will let you know if I stumble upon them.  I just skimmed the link you sent, and it looks like it raises some good points.  My motivation for pushing the PR-based workflow is to make it much easier to validate (via Travis) contributions.  By finding issues at the PR stage, we can avoid problems entering the codebase; pushing the new CC transform code as a PR would have prevented that simple missing "return" statement from affecting any users, for example.  That turned out to be a non-issue, but it serves as a nice example of how apparently good code can still introduce problems that can be difficult to hunt down without our new workflow.  The PR mechanism just serves to utilize the contributor's fork as a staging area for validation, which takes a lot of responsibility off the shoulders of the maintainer that clicks that "accept" button.  A very nice side effect of all of this, is that is brings us in line with most projects on GH, and lowers the barrier to getting contributions from outside the usual suspects.

**Comment by loriab** [comment]
I like the article @dgasmith linked to for the overall workflow. But the one I keep revisiting for how to do branches and PR is http://blog.jasonmeridth.com/posts/do-not-issue-pull-requests-from-your-master-branch/ (which is also linked to in the wiki). So with ``others`` and ``[good idea](http://blog.jasonmeridth.com/posts/do-not-issue-pull-requests-from-your-master-branch/)`` changes to CONTRIBUTING.md referenced above, I like the state of this PR.

**Comment by jturney** [comment]
I like that one @loriab. I'm going to start doing my PRs from branches and not from master. That way I can continue to work on code while I want on a PR to be accepted.

**Comment by andysim** [comment]
Good call @loriab .  I was about to add the text, but @dgasmith was more alert and beat me to it.

**Comment by dgasmith** [comment]
I think this is ready for a merge, does anyone have any other issues to corrections to make?

**Comment by andysim** [comment]
Looks good to me.  Thanks for everybody's helpful contributions.

**Pull request merged**